### PR TITLE
fix MSVC compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
the endless battle...

### CODE REVIEW

**Comment by Howard-C** [comment]
MSVC?

**Comment by Jojo-Schmitz** [comment]
MicroSoft Visual C++

**Comment by Howard-C** [comment]
> MicroSoft Visual C++, the compiler officially used for the Windows version

Yes, so it should be MSVC, not MSCV ;-)

File: mscore/palette/palettetree.cpp [hunk 1]
```diff
@@ -482,8 +482,8 @@ static void writePaletteFailed(const QString& path)
 bool PalettePanel::writeToFile(const QString& p) const
       {
       QSet<ImageStoreItem*> images;
-      int n = cells.size();
-      for (int i = 0; i < n; ++i) {
+      size_t n = cells.size();
```
Comment (anatoly-os) [inline mscore/palette/palettetree.cpp]: Using `auto` is the solution.

File: mscore/palette.cpp [hunk 2]
```diff
@@ -107,8 +107,8 @@ Palette::Palette(std::unique_ptr<PalettePanel> pp, QWidget* parent)
       setDrawGrid(pp->drawGrid());
       setMoreElements(pp->moreElements());
 
-      const auto cells = pp->takeCells(0, pp->ncells());
-      for (const PaletteCellPtr& cell : cells) {
+      const auto cs = pp->takeCells(0, pp->ncells());
```
Comment (anatoly-os) [inline mscore/palette.cpp]: I would call it `allPaletteCells` or `allCells`.

File: mscore/palette/palettetree.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply mscore/palette/palettetree.cpp]: Hmm, maybe. I'm still way to much of a C person I guess

Comment (Jojo-Schmitz) [reply mscore/palette/palettetree.cpp]: And no, it is not: it then leads to a warning in MinGW

File: mscore/palette.cpp (hunk 2 above)
Comment (Jojo-Schmitz) [reply mscore/palette.cpp]: OK

**Pull request merged**